### PR TITLE
Feat: web mapping color bars

### PR DESCRIPTION
**Pull request by MattBuckley-TfN**
# Changes

Added optional categorical parameter to ExploreOptions to select between categorical / continuous color mapping. Added white background to color bars to make text more visible.

Added a couple of issues (#42 and #43) to improve the legends in the HTML maps.

# Tasks

- [ ] Have unittests been added (testing individual functions and their edge cases)
- [ ] Has documentation (including user guide) been updated
- [ ] Have new dependencies been added (or changed) in relevant `requirements.txt`
- [ ] Code linting, tests and other workflows pass

### CODE REVIEW

File: src/caf/viz/web/mapping.py [hunk 1]
```diff
@@ -127,6 +136,9 @@ class ExploreOptions:
     legend_title: str | None = None
     cmap: str | list[str] = "viridis"
     show: bool = True
+    categorical: bool | None = None
+    """Whether the legend should be categorical or not.
+    If None defaults to False for numeric columns and true for other types."""
```
[the remark this answers is not in the document]
Comment (MattBuckley-TfN) [reply src/caf/viz/web/mapping.py]: This is incorrect sphinx picks up docstrings like this

File: src/caf/viz/web/mapping.py [hunk 2]
```diff
@@ -87,6 +88,14 @@
     .textbox-content {
         transition: all 0.3s ease;
     }
+    /* Continuous colorbar (Branca) */
+    .legend.leaflet-control:not(:empty) {
```
[the remark this answers is not in the document]
Comment (MattBuckley-TfN) [reply src/caf/viz/web/mapping.py:92]: This is wrong

File: src/caf/viz/web/mapping.py [hunk 3]
```diff
@@ -162,9 +174,17 @@ def _explore(
     else:
         legend = {}
 
+    if options.categorical is None:
+        if data_column is None:
+            categorical = False
+        else:
+            categorical = not pd.api.types.is_numeric_dtype(data[data_column])
```
[the remark this answers is not in the document]
Comment (MattBuckley-TfN) [reply src/caf/viz/web/mapping.py]: Fixed in bfd216d6212de0e528bee883b6eec3b1b5e1a0dd

**Comment by amber-mertens-tfn** [comment]
I think this looks good, thanks for adding this!

**Review by amber-mertens-tfn — APPROVED** [review]
